unstructured/partition/ocr.py

```diff
@@ -347,6 +354,53 @@ def init_table_agent():
     return table_agent
 
 
+def get_page_layout_from_ocr(
+    image: PILImage,
+    page_number: int = 1,
+    ocr_languages: str = "eng",
+    ocr_agent: str = OCR_AGENT_TESSERACT,
+) -> "PageLayout":
+    """
+    Generate a PageLayout with OCR data from a given image.
+    """
+
+    ocr_layout = get_ocr_data_from_image(
+        image,
+        ocr_languages=ocr_languages,
+        ocr_agent=ocr_agent,
+        output_type=OCROutputType.TEXT_REGIONS,
+    )
+
+    if ocr_agent == OCR_AGENT_PADDLE:
```

Comment:
i am not sure why we need to separate the logic for paddle and tesseract here

Reply:
For **tesseract**, the `ocr_text` returned by `unstructured_pytesseract.image_to_string()` doesn't contain bounding box data but is well grouped. Conversely, the `ocr_layout` returned by parsing `unstructured_pytesseract.image_to_data()` contains bounding box data but is not well grouped. Therefore, we need to first group the `ocr_layout` by `ocr_text` and then merge the text regions in each group to create a list of layout elements.
For **paddle**, there is no difference in `ocr_layout` and `ocr_text` in terms of grouping because we get `ocr_text` from `ocr_layout, so the first two grouping and merging steps are not necessary.

Reply:
gotcha good point! could you note this in the function? Thanks

Reply:
Those two methods are combined in `get_elements_from_ocr_regions()`, which is what I did in this PR.